### ISSUE TITLE
Consider to use version ranges, follow semver

### ISSUE THREAD

**Issue by langpavel** [assetgraph/assetgraph#511]
Hi man, I see your work and you have too many issues maintaining this project.. But it is really unnecessary. You already using `david-dm`, `travis`, `coveralls`, but why? When someone comes on this project it will see all red and bad?

You really should switch to semver model. You have tests, haven't you? When new package of any dependency appears, `greenkeeper` will report you if test fail. When something weird is discovered in third party package, don't bother, they will fix it in patch version and you will be free too. At the moment `david-dm` is reporting serious vulnerability in uglify. Do you really want to lock package version known for it's vurneability in *all the projects yousing yours*?

Much less annoying messages and only important will match you if you use version ranges. Of course, you always can (and you should) watch whats happening... GitHub notifications are there for this purpose.. It's reason why I write you this. I just receive much notifications for your package.

I know you maintain this well and I wish you much happy users :-)
If you feel that something of your dependencies should be watched more carefully, you can, but just try to specify ranges (or better, semver ranges) on packages which are widely used. Because of pretty behavior of `npm` your package can prevent others to upgrade just because of strict version requirement of yours.

**Comment by papandreou** [comment]
I agree that it looked a bit like a train wreck. The fact that we use pinned versions stems from old trust issues and times where we had worse tooling, so I think you're right that it's time to loosen up. I've changed a bunch of dependencies that have a good track record to use `^` and `~` version ranges now. Let's see how it goes.

Specifically, the reason why we kept using an old version of UglifyJS was that they introduced a regression somewhere in 2.5.x. I now took the time to [hunt down and report it](https://github.com/mishoo/UglifyJS2/pull/810#issuecomment-216073186) and untimately work around the problem so we're now on the newest version. A similar thing was the case with html-minifier: https://github.com/kangax/html-minifier/issues/215#issuecomment-216070213

The lesson from the two above cases is that it takes a continuous effort to keep up with all your dependencies, especially with a test suite as involved as assetgraph's. I don't want to get into a situation where it fails every other day and thus gets in the way of actual development.
